src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +88,145 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        this.loadAndMergeOriginData(this.schema, schema).then(action(()=> {
+            this.removeObsoleteData(this.schema, schema, this.data);
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.setSchemaLoading(false);
+        }));
     };
 
+    loadAndMergeOriginData(oldSchema: Schema, newSchema: Schema) {
+        if (oldSchema && toJS(this.type) === this.data['originTemplate']){
+            return this.resourceStore.loadData().then((data: Object) => {
+                this.mergeData(oldSchema, newSchema, this.data, data);
```

Comment:
i think i would expect this method to return a new data array and not change the data of the store

Reply:
if the method returns a new array that only contains the keys of the new schema, we might also not need the `removeObsoleteData`?